packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -146,9 +159,6 @@ export let visitor = {
 
       // whether any references to the rest parameter were made in a function
       deopted: false,
-
-      // whether all we need to do is replace rest parameter identifier with 'arguments'
-      replaceOnly: false
```

Comment:
@vhf I removed the `replaceOnly` and categorized each optimisation, using the `cause` key.